### ISSUE TITLE
MW 1.37: ParserOptions constructor must be called with a UserIdentity object

### ISSUE THREAD

**Issue by morganfletcher** [SemanticMediaWiki/SemanticMediaWiki#5177]
_Which version of Semantic MediaWiki is used?_
4.0.0-rc1 
Using 'dev-master', just did php /opt/bitnami/php/bin/composer.phar update --no-dev ; php maintenance/update.php

_Which version of a related semantic extensions is used? (If applicable.)_
Semantic Result Formats 3.2.0
Similarly, 'dev-master'.

_Which version of MediaWiki is used?_
1.37.0

_Which server setup was used (PHP, MySQL, Operating System)?_
[Bitnami GCP image](https://bitnami.com/stack/mediawiki/cloud/google)

_Can the problem be viewed online in a public wiki? (If yes, provide its URL.)_
https://www.wooljersey.com/

_Can the problem be reproduced on a test page on a sandbox site? (If yes, provide its URL.)_
I don't know.

```
[824f52ff3a8a19135d2dca54] /index.php?title=THE_WHEELMEN._-_The_San_Francisco_Call,_31_Aug_1895&action=submit TypeError: Argument 1 passed to ParserOptions::__construct() must implement interface MediaWiki\User\UserIdentity, null given, called in /bitnami/mediawiki/extensions/SemanticMediaWiki/includes/ContentParser.php on line 198

Backtrace:

from /opt/bitnami/mediawiki/includes/parser/ParserOptions.php(1057)
#0 /bitnami/mediawiki/extensions/SemanticMediaWiki/includes/ContentParser.php(198): ParserOptions->__construct()
#1 /bitnami/mediawiki/extensions/SemanticMediaWiki/includes/ContentParser.php(137): SMW\ContentParser->makeParserOptions()
#2 /bitnami/mediawiki/extensions/SemanticMediaWiki/includes/ContentParser.php(126): SMW\ContentParser->parseText()
#3 /bitnami/mediawiki/extensions/SemanticMediaWiki/src/Factbox/CachedFactbox.php(337): SMW\ContentParser->parse()
#4 /bitnami/mediawiki/extensions/SemanticMediaWiki/src/Factbox/CachedFactbox.php(215): SMW\Factbox\CachedFactbox->rebuild()
#5 /bitnami/mediawiki/extensions/SemanticMediaWiki/src/MediaWiki/Hooks/OutputPageParserOutput.php(141): SMW\Factbox\CachedFactbox->prepare()
#6 /bitnami/mediawiki/extensions/SemanticMediaWiki/src/MediaWiki/Hooks/OutputPageParserOutput.php(100): SMW\MediaWiki\Hooks\OutputPageParserOutput->addFactbox()
#7 /bitnami/mediawiki/extensions/SemanticMediaWiki/src/MediaWiki/Hooks.php(461): SMW\MediaWiki\Hooks\OutputPageParserOutput->process()
#8 /opt/bitnami/mediawiki/includes/HookContainer/HookContainer.php(338): SMW\MediaWiki\Hooks->onOutputPageParserOutput()
#9 /opt/bitnami/mediawiki/includes/HookContainer/HookContainer.php(137): MediaWiki\HookContainer\HookContainer->callLegacyHook()
#10 /opt/bitnami/mediawiki/includes/HookContainer/HookRunner.php(2697): MediaWiki\HookContainer\HookContainer->run()
#11 /opt/bitnami/mediawiki/includes/OutputPage.php(1980): MediaWiki\HookContainer\HookRunner->onOutputPageParserOutput()
#12 /opt/bitnami/mediawiki/includes/EditPage.php(4119): OutputPage->addParserOutputMetadata()
#13 /opt/bitnami/mediawiki/includes/EditPage.php(2893): EditPage->getPreviewText()
#14 /opt/bitnami/mediawiki/includes/EditPage.php(722): EditPage->showEditForm()
#15 /opt/bitnami/mediawiki/includes/actions/EditAction.php(71): EditPage->edit()
#16 /opt/bitnami/mediawiki/includes/actions/SubmitAction.php(38): EditAction->show()
#17 /opt/bitnami/mediawiki/includes/MediaWiki.php(538): SubmitAction->show()
#18 /opt/bitnami/mediawiki/includes/MediaWiki.php(320): MediaWiki->performAction()
#19 /opt/bitnami/mediawiki/includes/MediaWiki.php(925): MediaWiki->performRequest()
#20 /opt/bitnami/mediawiki/includes/MediaWiki.php(559): MediaWiki->main()
#21 /opt/bitnami/mediawiki/index.php(53): MediaWiki->run()
#22 /opt/bitnami/mediawiki/index.php(46): wfIndexMain()
#23 {main}
```

**Comment by malberts** [comment]
This should be fixed as part of various test fixes in #5178 (https://github.com/SemanticMediaWiki/SemanticMediaWiki/pull/5178/files#diff-694a7ee0c343cfbbce308ff6318c10da351cc59cbd4aafb7c754c95e51b771a6R199).

MW 1.35 and 1.36 allowed passing in a null user which would then default to the current user, however that fallback was deprecated in MW 1.36 and removed in MW 1.37:
https://github.com/wikimedia/mediawiki/blob/REL1_36/includes/parser/ParserOptions.php#L1107

**Comment by morganfletcher** [comment]
Confirmed the fix, thank you.
